homeassistant/helpers/intent.py

```diff
@@ -110,7 +129,27 @@ def async_validate_slots(self, slots):
     @asyncio.coroutine
     def async_handle(self, intent_obj):
         """Handle the intent."""
-        raise NotImplementedError()
+        if not self.service:
+            raise NotImplementedError()
+        hass = intent_obj.hass
+        slots = self.async_validate_slots(intent_obj.slots)
+        name = slots['name']['value']
+        entity = _match_entity(hass, name)
```

Comment:
This is retrieving a state not an entity.

Reply:
It;s a roundabout way of retrieving all entites, by getting all states and using the entities that belong to them. Need to add filtering by domain.

Reply:
The state machine doesn't store entities, it stores states. You should change the variable name to reflect what it is.

Reply:
The function returns the entity_id though and not the state.

Reply:
```return hass.states.get(entity_id) if entity_id else None```

Reply:
No, it returns a state:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/core.py#L662-L668

Reply:
Ah, I see what you mean. I didn't write that bit. It would be more confusing to return a state, so rewrote the return

Reply:
Let's rename it to entity_id, as that is what is returned from `_match_entity`.